CRM/Contact/Import/Form/MapField.php

```diff
@@ -803,44 +779,25 @@ public function submit($params, $mapperKeys) {
           $relationType = new CRM_Contact_DAO_RelationshipType();
           $relationType->id = $id;
           $relationType->find(TRUE);
-          $relatedContactTypeVal = $relationType->{"contact_type_$second"};
-          $relatedContactDetailsVal = $this->_formattedFieldNames[$relatedContactTypeVal][$selOne];
+          $parserParameters['relatedContactType'][$i] = $relationType->{"contact_type_$second"};
+          $parserParameters['relatedContactDetails'][$i] = $this->_formattedFieldNames[$parserParameters['relatedContactType'][$i]][$selOne];
         }
       }
-
-      //set the respective mapper param array values.
-      foreach ($mapperParams as $mapperParamKey => $mapperParamVal) {
```

Comment:
The point of this is to transfer the value e.g $relatedContactTypeVal into the $relatedContactType array - weird huh!